docs/authentication.md

```diff
@@ -0,0 +1,155 @@
+# Authentication
+
+This document details how to configure your Helios installation to authenticate
+requests between the Helios CLI and the Helios masters/cluster.
+
+The support for authentication in Helios is pluggable; the specific scheme used
+by an installation can be controlled based on the arguments that Helios is
+started with. New authentication schemes can be added to Helios via
+implementing a handful of Java interfaces, packaging your plugin as a jar, and
+supplying the path to that JAR in an argument when starting the Helios cluster.
+
+Helios comes with support for the [crtauth protocol][crtauth-protocol], which
+is in use internally at Spotify, but users of Helios are free to choose any
+other scheme.
+
+# TODOs for this document
+
+- link to how to setup HTTPS for Helios masters
+
+## Authentication flow
+
+*Note that since all authentication requires clients to send some sort of
+credentials over the wire to the Helios masters, you should first configure
+your Helios clusters to use HTTPS.*
```

Comment:
I wonder, should we make it so the client just gives up and throws an exception if it's asked to authenticate over a plain HTTP connection?